gateway/handler.go

```diff
@@ -313,6 +314,13 @@ func (i *handler) getOrHeadHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Validate the root CID before any backend calls
+	rootCid := rq.immutablePath.RootCid()
+	if err := verifcid.ValidateCid(verifcid.DefaultAllowlist, rootCid); err != nil {
```

Comment:
Why does this need to happen before / what's the bad thing that happens if we rely on the blockservice, etc. used by the backend the user has passed in?

Asking both to understand and because the more places we hardcode globals like verifcid.DefaultAllowlist the harder / more hidden complexity we put into the system.

Reply:
just a precaution while i write tests.

before making this PR ready for review i'll remove it and see if blockstore is enough